File: src/Software Engineer Prepare Kit/16/solution.go
package main

import "maps"

func IsAnagram(s string, t string) int32 {
	if len(s) != len(t) {
		return 0
	}

	sChars := make(map[byte]int16)
	tChars := make(map[byte]int16)

	for i := 0; i < len(s); i++ {
		sChars[s[i]] += 1
		tChars[t[i]] += 1
	}

	if maps.Equal(sChars, tChars) {
		return 1
	}

	return 0
}

File: src/Software Engineer Prepare Kit/16/solution.ts
export default function isAnagram(s: string, t: string): number {
	if (s.length !== t.length) return 0

	const sChars = new Map<string, number>()
	const tChars = new Map<string, number>()

	for (let i = 0; i < s.length; i++) {
		const sChar = s[i]!
		const tChar = t[i]!

		sChars.set(sChar, (sChars.get(sChar) ?? 0) + 1)
		tChars.set(tChar, (tChars.get(tChar) ?? 0) + 1)
	}

	if (sChars.size !== tChars.size) return 0

	for (const [char, amount] of sChars) {
		if (amount !== tChars.get(char)) return 0
	}

	return 1
}
